bin/hdfcoinc/pycbc_coinc_statmap

```diff
@@ -146,7 +219,160 @@ else:
     f['foreground/ifar_exc'] = numpy.array([])
     f['foreground/fap_exc'] = numpy.array([])
 
-if 'name' in d.attrs:
-    f.attrs['name'] = d.attrs['name']
+if 'name' in all_trigs.attrs:
+    f.attrs['name'] = all_trigs.attrs['name']
+
+# Incorporate hierarchical removal for any other loud triggers
+logging.info("Instantiating hierarchical removal.")
+
+# Step 1: Check to see if the trigger with the largest FAR is
+#         louder than all of the background.
+# fnlouder is the number of background events that are louder than
+# the foreground trigger. So if a trigger is louder than all of
+# the background, fnlouder = 0.
+
+loud_t_ind = []
+count_ind = -1
+for i in range(0, len(fnlouder)):
+    if fnlouder[i] == 0:
+        loud_t_ind.append(i)
+        count_ind += 1
+
+# Step 2 : Loop until we don't have to hierarchically remove
+#          anymore.
+iter = 1
+while len(loud_t_ind) > 0 and iter <= args.max_hier_removal :
+    # find the loudest of these triggers, its index, its newsnr
+    # and its ifar
+    max_ifar = 0
+    max_ifar_ind = 0
+    max_rank_stat = 0
+    for i in range(0, len(loud_t_ind)):
+         # Check for the largest ifar
+         if ifar[loud_t_ind[i]] >= max_ifar:
+           # Check to see if they have equal ifar, then check largest
+           # ranking statistic
+              if ifar[loud_t_ind[i]] == max_ifar:
+                   if back_stat[loud_t_ind[i]] >= \
+                      back_stat[loud_t_ind[max_ifar_ind]] :
+                        max_ifar_ind = loud_t_ind[i]
+                        max_rank_stat = back_stat[loud_t_ind[i]]
+                   else :
+                        max_rank_stat = back_stat[max_ifar_ind]
+                        
+              else :
+                   max_ifar = ifar[loud_t_ind[i]]
+                   max_ifar_ind = loud_t_ind[i]
+    # Step 3: Remove that trigger from the list of zerolag triggers
+
+    # FIXME:If the trigger is really the loudest trigger in the bin, then it's
```

Comment:
Hey @stevereyes01 @ahnitz, I think this works in the case and there is no fix me required. If people agree, you might want to remove this comment and print statement from the code. In the least bit, use logging instead of ``print``.

But here's why I think what Steven has written works for the case he's worried about. If you have two triggers with the same detection, no matter which one you remove, then next one will still always be larger and removed next. And this will mostly likely be a very rare thing.

I think what you need to add is some ordering to ``where_is_fore_t`` so its always deterministic which one is removed first. That way if the code is rerun with different ``--max-heir-removal`` many times, you always get the same.

I'd suggest ordering it. And in the caption give a ordered list of the GPS times that the triggers were removed.